microsoft/testsuites/core/cpu.py

```diff
@@ -19,6 +27,47 @@
     """,
 )
 class CPU(testsuite.TestSuite):
+    @testsuite.TestCaseMetadata(
+        description="""
+            This test will check that CPU assigned to lsvmbus
+            channels cannot be put offline.
+            Steps :
+            1. Get the list of lsvmbus channel cpu mappings using
+            command `lsvmbus -vv`.
+            2. Create a set of cpu's assigned to lsvmbus channels.
+            3. Try to put cpu offline by running
+            `echo 0 > /sys/devices/system/cpu/cpu/<cpu_id>/online`.
+            Note : We skip cpu 0 as it handles system interrupts.
+            4. Ensure that cpu is still online by checking state '1' in
+            `/sys/devices/system/cpu/cpu/<target_cpu>/online`.
+            """,
```

Comment:
Bring CPUs online after tested. Other test cases may run on the same node.

Reply:
Is it possible to delete the environment associated with this test? The test could potentially take the environment into a bad state if it fails to come up online.

Reply:
Yes, it's possible. We should have another exception to make sure recycle the environment, if a test case declare the environment is in bad state. But anyway, we should try to recover this environment after test case completed.